tablecodec/tablecodec.go

```diff
@@ -1300,21 +1300,21 @@ func TruncateIndexValue(v *types.Datum, idxCol *model.IndexColumn, tblCol *model
 	if notStringType {
 		return
 	}
-	originalKind := v.Kind()
-	isUTF8Charset := tblCol.Charset == charset.CharsetUTF8 || tblCol.Charset == charset.CharsetUTF8MB4
-	if isUTF8Charset && utf8.RuneCount(v.GetBytes()) > idxCol.Length {
-		rs := bytes.Runes(v.GetBytes())
+	colValue := v.GetBytes()
+	if tblCol.Charset == charset.CharsetBin || tblCol.Charset == charset.CharsetASCII {
```

Comment:
We change this since we are adding the GBK encoding  and it would actually encoding as `UTF8` then go into the `else ifutf8.RuneCount(colValue) > idxCol.Length` right?
I think we can comment this to help others understand the code.

Reply:
@xiongjiwei Please address this comment

Reply:
the comments are added.